File: src/main/java/net/hectus/neobb/turn/default_game/structure/TDaylightSensorLine.java
package net.hectus.neobb.turn.default_game.structure;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.structure.Structure;
import net.hectus.neobb.structure.StructureManager;
import net.hectus.neobb.turn.default_game.CounterFilter;
import net.hectus.neobb.turn.default_game.attributes.clazz.HotClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.NatureClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.RedstoneClazz;
import net.hectus.neobb.turn.default_game.attributes.clazz.WaterClazz;
import net.hectus.neobb.turn.default_game.attributes.function.CounterbuffFunction;
import net.hectus.neobb.util.MinecraftTime;

import java.util.List;

public class TDaylightSensorLine extends StructureTurn implements CounterbuffFunction, RedstoneClazz {
    public TDaylightSensorLine(NeoPlayer player) { super(player); }
    public TDaylightSensorLine(PlacedStructure data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 6;
    }

    @Override
    public Structure referenceStructure() {
        return StructureManager.structure("daylight_sensor_line");
    }

    @Override
    public void apply() {
        player.game.time(MinecraftTime.DAY);
    }

    @Override
    public List<Buff> buffs() {
        return List.of();
    }

    @Override
    public List<CounterFilter> counters() {
        return List.of(CounterFilter.clazz(HotClazz.class), CounterFilter.clazz(NatureClazz.class), CounterFilter.clazz(WaterClazz.class), CounterFilter.clazz(RedstoneClazz.class));
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/structure/TDaylightSensorLine.kt
package net.hectus.neobb.turn.default_game.structure

import com.marcpg.libpg.storing.Cord
import com.marcpg.libpg.util.MinecraftTime
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.structure.StaticStructure
import net.hectus.neobb.structure.StaticStructures
import net.hectus.neobb.turn.default_game.CounterFilter
import net.hectus.neobb.turn.default_game.attribute.clazz.HotClazz
import net.hectus.neobb.turn.default_game.attribute.clazz.NatureClazz
import net.hectus.neobb.turn.default_game.attribute.clazz.RedstoneClazz
import net.hectus.neobb.turn.default_game.attribute.clazz.WaterClazz
import net.hectus.neobb.turn.default_game.attribute.function.CounterbuffFunction

class TDaylightSensorLine(data: PlacedStructure?, cord: Cord?, player: NeoPlayer?) : StructureTurn(data, cord, player), CounterbuffFunction, RedstoneClazz {
    override val cost: Int = 6
    override val staticStructure: StaticStructure = StaticStructures.Default.DAYLIGHT_SENSOR_LINE

    override fun apply() {
        player!!.game.time = MinecraftTime.DAY
    }

    override fun counters(): List<CounterFilter> {
        return listOf(
            CounterFilter.clazz(HotClazz::class),
            CounterFilter.clazz(NatureClazz::class),
            CounterFilter.clazz(WaterClazz::class),
            CounterFilter.clazz(RedstoneClazz::class)
        )
    }

    override fun buffs(): List<Buff<*>> = emptyList()
}
